### DISCUSSION TITLE
一直搞不清楚為什麼字典是沒有順序的資料型態

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#154] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/86505454/141668412-873d6002-ae96-4841-a5e7-7161e17871fb.png)
一直搞不懂為什麼字典是沒有順序的資料型態。我的dict1與dict2兩個字典都打一樣的東西，但為什麼dict1與dict2顯示出來的結果不一樣。執行dict1與dict2的結果如果不一樣，不就代表dict1與dict2字典的資料型態，是有順序性的嗎？

**Comment by chia-kai-chang** [comment, marked as answer]
即使排列順序不同，但只要內容相同，在記憶體層面上，就是同一個物件。可以參考下圖的程式碼。

**順序**對於**字典**來說是**無關緊要的**。

![image](https://user-images.githubusercontent.com/1768891/141672671-f1440ddc-56aa-4626-bb89-23ae068e897b.png)

**Reply by enoch-lightning** [reply]
原來使用a==b(==是比較運算子)來驗證a與b的值(value)是否相同，就可知道此資料型態有沒有順序性。

**Reply by enoch-lightning** [reply]
![image](https://user-images.githubusercontent.com/86505454/149065227-ab4189b5-953f-481a-beb1-3f5914a415f4.png)
由上圖可知 ，雖a==b為True，代表a與b的值相同。但，從a is b或id(a)==id(b)均可看出a與b的指向的字典物件的記憶體位置不同。代表所謂是否有順序，指的是組成物件的三個要素—identity(記憶體位置)、type(資料型態)與value(值)中的「value(值)」是否有順序。

**Reply by enoch-lightning** [reply]
Python的原生語法中，sequence types、text sequence type與binary sequence types這些名字中有sequence的資料型態，就是有順序的資料型態。
